### ISSUE TITLE
Connection loss detection

### ISSUE THREAD

**Issue by zerodivisi0n** [mariano/node-db-mysql#11]
Is there a way to detect loss of connection to the server? How about a use for this purpose event <code>'error'</code> on <code>Database</code> object?

And also use option <code> MYSQL_OPT_RECONNECT</code> when executing queries to automatic reconnection.

**Comment by mariano** [comment]
Let me think about the loss detection for a bit.

Regarding the reconnect, there is a setting (amongst others) for it since db-mysql 0.5.7. Specifically set the option reconnect to true when creating the Database instance. Refer to the [db-mysql connection documentation](http://nodejsdb.org/docs/#database_db-mysql) for this and other options.

**Comment by zerodivisi0n** [comment]
Oh! Thank you for advice about the options. Had not noticed them before.

Your project is developing rapidly. Keep it up!

**Comment by mariano** [comment]
This has been fixed in db-mysql 0.6.2 by doing the following:
1. The reconnect option is enabled by default
2. The Database.isConnected() method properly pings the server to check status of connection

Check the [release announcement](http://nodejsdb.org/2011/05/node-db-0-6-2-released/) and update to the latest db-mysql release

**Comment by zerodivisi0n** [comment]
I think that pinging the server is not a best solution for checking connection. Is whether a way to emit event when the connection is dropping?

**Comment by mariano** [comment]
The pinging _only happens_ when you issue Database.isConnected(). Internally I'm not doing anything with pinging, it would make no sense to do so.

As far as I know there is no way in the MYSQL API to be informed of a dropped connection. That's why the 'reconnect' setting is there: to automatically reconnect in the event of a connection drop. If MySQL was able to tell you when it dropped, then reconnect would be moot.

As per the MySQL API, ping is the way to determine if a connection is alive. The only other alternative is to check the thread_id() of the connection against the previous thread_id(). If it's different, it means a reconnection occured. But that just serves no purpose other than irrelevant information, imho

**Comment by zerodivisi0n** [comment]
I've got the following error when trying to use <code>reconnect</code> flag:

```
> var db_mysql = require('db-mysql')
> var db = new db_mysql.Database({reconnect: true})
node: symbol lookup error: /home/zerodivisi0n/node/lib/node_modules/db-mysql/build/default/mysql_bindings.node: undefined symbol: _ZN13node_db_mysql10Connection12setReconnectEb
```

**Comment by mariano** [comment]
Fixed in ef347f2739384a89851fa27ffacacbfcf317d42e

Update to latest db-mysql. [View 0.6.4 release announcement](http://nodejsdb.org/2011/06/db-mysql-0-6-4-released/)
